app/dao/notifications_dao.py

```diff
@@ -433,18 +379,23 @@ def insert_notification_history_delete_notifications(
     return result
 
 
-def _move_notifications_to_notification_history(notification_type, service_id, day_to_delete_backwards_from, qry_limit):
+def move_notifications_to_notification_history(
+    notification_type,
+    service_id,
+    timestamp_to_delete_backwards_from,
+    qry_limit=50000
+):
     deleted = 0
     if notification_type == LETTER_TYPE:
         _delete_letters_from_s3(
-            notification_type, service_id, day_to_delete_backwards_from, qry_limit
+            notification_type, service_id, timestamp_to_delete_backwards_from, qry_limit
         )
     delete_count_per_call = 1
     while delete_count_per_call > 0:
```

Comment:
if we wanted to be really visibility timeout-concious, we could split into tasks at this level.

ie: "delete 50,000 notifications. If there's potentially more to delete, then enqueue a new task to delete the next batch of 50,000".

That would ensure all the tasks were a bit smaller, and much much more likely to stay under 5 minutes.

I don't think we should do that now, and the complexity could be a bit hard to approach, but it's a potential next step if we really wanted to keep task durations lower.